NuRadioMC/SignalProp/analyticraytracing.py

```diff
@@ -153,215 +431,61 @@ def __init__(self, medium, attenuation_model="SP1",
         if overwrite_speedup is not None:
             self._use_optimized_calculation = overwrite_speedup
         self.use_cpp = use_cpp
+        if compile_numba:
+            if numba_available:
+                global get_reflection_point,obj_delta_y_square,get_delta_y
+                global get_y_turn, get_y_with_z_mirror,get_turning_point
+                global get_gamma, get_y, get_C0_from_log
+                global  get_z_unmirrored, get_y_diff, n
```

Comment:
good point, but is it possible to disable NUMBA without re-initializing the class? I think no. Then your concern could never happen. 

And it would make all the rest of the code more messy if it is filled with if/else statements whenever one of the compiled functions is called.